mmgen/models/architectures/biggan/generator_discriminator.py

```diff
@@ -0,0 +1,532 @@
+                'attention': [False, False, False, False]
+            },
+            '64': {
+                'in_channels': [in_channels] +
+                [base_channels * item for item in [1, 2, 4, 8]],
+                'out_channels':
+                [base_channels * item for item in [1, 2, 4, 8, 16]],
+                'downsample': [True] * 4 + [False],
+                'resolution': [32, 16, 8, 4, 4],
+                'attention': [False, False, False, False, False]
+            },
+            '128': {
+                'in_channels': [in_channels] +
+                [base_channels * item for item in [1, 2, 4, 8, 16]],
+                'out_channels':
+                [base_channels * item for item in [1, 2, 4, 8, 16, 16]],
+                'downsample': [True] * 5 + [False],
+                'resolution': [64, 32, 16, 8, 4, 4],
+                'attention': [True, False, False, False, False, False]
+            },
+            '256': {
+                'in_channels': [in_channels] +
+                [base_channels * item for item in [1, 2, 4, 8, 8, 16]],
+                'out_channels':
+                [base_channels * item for item in [1, 2, 4, 8, 8, 16, 16]],
+                'downsample': [True] * 6 + [False],
+                'resolution': [128, 64, 32, 16, 8, 4, 4],
+                'attention': [False, True, False, False, False, False]
+            },
+            '512': {
+                'in_channels': [in_channels] +
+                [base_channels * item for item in [1, 1, 2, 4, 8, 8, 16]],
+                'out_channels':
+                [base_channels * item for item in [1, 1, 2, 4, 8, 8, 16, 16]],
+                'downsample': [True] * 7 + [False],
+                'resolution': [256, 128, 64, 32, 16, 8, 4, 4],
+                'attention': [False, False, False, True, False, False, False]
+            }
+        }
+
+        return _default_arch_cfgs[str(input_scale)]
+
+    def forward(self, x, label=None):
+        # TODO:
+        """[summary]
+
+        Args:
+            x ([type]): [description]
+            label ([type], optional): [description]. Defaults to None.
+
+        Returns:
+            [type]: [description]
+        """
+        x0 = x
+        for conv_block in self.conv_blocks:
+            x0 = conv_block(x0)
+        x0 = self.activation(x0)
+        x0 = torch.sum(x0, dim=[2, 3])
+        out = self.linear(x0)
+
+        if self.num_classes > 0:
+            w_y = self.proj_y(label)
+            out = out + torch.sum(w_y * x0, dim=1, keepdim=True)
+        return out
+
+    def init_weights(self, pretrained=None, init_type='ortho'):
+        """Init weights for models.
+
+        We just use the initialization method proposed in the original paper.
+
+        Args:
+            pretrained (str, optional): Path for pretrained weights. If given
+                None, pretrained weights will not be loaded. Defaults to None.
+        """
+
+        if isinstance(pretrained, str):
```

Comment:
We may support load pre-trained model like this:
https://github.com/open-mmlab/mmgeneration/blob/master/mmgen/models/architectures/stylegan/generator_discriminator_v2.py#L211

Because there are many works using pretrained BigGAN generator and BigGAN discriminator.